xwiki-rendering-macros/xwiki-rendering-macro-message/src/main/java/org/xwiki/rendering/internal/macro/message/AbstractMessageMacro.java

```diff
@@ -73,4 +100,43 @@ public void prepare(MacroBlock macroBlock) throws MacroPreparationException
     {
         this.contentParser.prepareContentWiki(macroBlock);
     }
+
+    @Override
+    public List<Block> execute(BoxMacroParameters parameters, String content, MacroTransformationContext context) 
+        throws MacroExecutionException 
+    {
+        List<Block> boxFoundation = super.execute(parameters, content, context);
+        if (!boxFoundation.isEmpty() && this.iconName != null) {
+            Block defaultBox = boxFoundation.get(0);
+            // For an easier styling, we wrap the content and title together if they are non-empty and visible
+            if (defaultBox.getChildren().size() > 1) {
+                Block boxTextContent = new GroupBlock(defaultBox.getChildren());
+                defaultBox.setChildren(List.of(boxTextContent));
+            }
+            // Enhance the default box with an icon as the first element.
+            Block iconBlock = this.iconProvider.get(this.iconName);
+            // Provide an accessible name besides this icon
+            // This is the responsibility of the message macro and not the iconProvider which should only provide
+            // icons without any semantics
+            String iconPrettyName = iconPrettyNameProvider.getIconPrettyName(this.iconName);
```

Comment:
I'm wondering if it would not be much simpler and more logical anyway to just pass the macro id here instead of having to go through a "iconName" variable ?

Also, it would probably make more sense to put the `ApplicationResources.properties` filed with the translations on rendering side since that's where are located the macros those translations are about.

Reply:
> I'm wondering if it would not be much simpler and more logical anyway to just pass the macro id here instead of having to go through a "iconName" variable ?

I'm not sure what we could use as a macro id. The macro class would work since we want the same alternative for all the macros of the same class. Since this is internal, we can make the assumption : 1 macro type <-> 1 icon name

> Also, it would probably make more sense to put the ApplicationResources.properties file with the translations on rendering side since that's where are located the macros those translations are about.

Done in https://github.com/xwiki/xwiki-rendering/pull/296/commits/0a423cec12a96c1488e1f1f8a0f7bb7471a8e5c0 :+1: 
I'll keep in mind that we need to create a new weblate project when this PR is merged so that it's easy to contribute translations to this new file.

Reply:
> I'm not sure what we could use as a macro id.

I don't understand the question, each macro has an id, what is used as role hint.

Reply:
Added in https://github.com/xwiki/xwiki-rendering/pull/296/commits/055972f22dff4437e94aeefeafce54953cce957f that icon provided via this method are always inline. :+1: